lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -51,16 +53,24 @@ final class CachedReader implements Reader
     private $loadedAnnotations = [];
 
     /**
-     * Constructor.
-     *
-     * @param Reader $reader
-     * @param Cache  $cache
-     * @param bool   $debug
+     * @param Cache|CacheInterface $cache
+     * @param bool                 $debug
      */
-    public function __construct(Reader $reader, Cache $cache, $debug = false)
+    public function __construct(Reader $reader, $cache, $debug = false)
     {
+        if ($cache instanceof Cache) {
+            @trigger_error(sprintf(
+                'Passing a %s instance to %s is deprecated since doctrine/annotations 1.9 and will not be possible anymore in 2.0. Please pass a %s instance',
+                Cache::class,
+                __METHOD__,
+                CacheInterface::class
+            ), E_USER_DEPRECATED);
+            $cache = new SimpleCacheAdapter($cache);
```

Comment:
@Ocramius I need some guidance here, because the adapter will throw if 2 interfaces are not implemented: https://github.com/Roave/DoctrineSimpleCache/blob/165599198031cf2a14b52bc7de6eca970e9dfdf5/src/SimpleCacheAdapter.php#L27-L33 , which means this is a BC break

Reply:
Maybe these exception could be moved to the methods that require them?

Reply:
All our providers inherit from the abstract class, which implements all three interfaces. Maybe we should expect that abstraction instead of the interface?